localstack-core/localstack/services/kms/models.py

```diff
@@ -361,6 +365,27 @@ def verify(
             # AWS itself raises this exception without any additional message.
             raise KMSInvalidSignatureException()
 
+    def derive_shared_secret(self, public_key: bytes) -> bytes:
+        key_spec = self.metadata.get("KeySpec")
+        if key_spec in [KeySpec.ECC_NIST_P256, KeySpec.ECC_SECG_P256K1]:
+            algorithm = hashes.SHA256()
+        elif key_spec == KeySpec.ECC_NIST_P384:
+            algorithm = hashes.SHA384()
+        elif key_spec == KeySpec.ECC_NIST_P521:
+            algorithm = hashes.SHA512()
+        else:
+            raise InvalidKeyUsageException(
+                f"{self.metadata['Arn']} key usage is {self.metadata['KeyUsage']} which is not valid for DeriveSharedSecret."
+            )
```

Comment:
This would be a nice place for a `match case` statement I think, but it's fine and easily legible like this as well